packages/ui-kit/src/components/TimeRangePicker/TimeRangePicker.tsx

```diff
@@ -0,0 +1,454 @@
+import { ClickAwayListener } from '@mui/base/ClickAwayListener'
+import { Popper } from '@mui/base/Popper'
+import classNames from 'classnames'
+import { format, startOfDay } from 'date-fns'
+import React from 'react'
+import { ClassNames, DateRange, DayPicker } from 'react-day-picker'
+import styles from 'react-day-picker/dist/style.module.css'
+import { useForwardedRefCallback } from '../../helpers'
+import { Button } from '../Button'
+import { Divider } from '../Divider'
+import { CalendarIcon } from '../Icons/Calendar'
+import { CloseIcon } from '../Icons/Close'
+import { Input } from '../Input'
+import { Select } from '../Select'
+import { Option } from '../Select/Option'
+import { useSetupTheme } from '../ThemeProvider'
+import { Typography } from '../Typography'
+import { DateTimeField } from './DateTimeField'
+import {
+  CUSTOM_DATE_RANGE,
+  CUSTOM_RANGE_FORMAT,
+  DATE_FORMAT,
+  defaultOptions,
+  FROM_DATE_UNTIL_NOW,
+  lastNOptions
+} from './TimeRangePicker.const'
+import componentStyles from './TimeRangePicker.module.scss'
+import { DateRangeOptionsProps, TimeRangePickerProps } from './TimeRangePicker.types'
+
+const formatDateTime = (value: Date | undefined, valueFormat: string) => (value ? format(value, valueFormat) : '')
+
+export const TimeRangePicker = React.forwardRef<HTMLDivElement, TimeRangePickerProps>(
+  (
+    {
+      baseTheme,
+      disableDateUntilNow = false,
+      disableCustomRange = false,
+      disableCustomRelative = false,
+      disableOptions = false,
+      disabled = false,
+      options: optionsProp,
+      defaultValue: defaultValueProp,
+      value,
+      onChange
+    },
+    forwardedRef
+  ) => {
+    const { componentContainer, setRef } = useForwardedRefCallback(forwardedRef)
+    const { theme } = useSetupTheme({
+      baseTheme,
+      componentContainer
+    })
+    const options = React.useMemo(() => (optionsProp ? optionsProp(defaultOptions) : defaultOptions), [optionsProp])
+    const [defaultValue] = React.useState(defaultValueProp)
```

Comment:
I this necessary? `defaultValueProp` in derived to `defaultValue` and then `defaultValue` is derived to `selectedOption`, then `defaultValue` is not used anywhere else, can we get rid of it and just pass `defaultValueProp` to `selectedOption` instead?